packages/flutter_tools/lib/src/project.dart

```diff
@@ -467,6 +467,25 @@ class IosProject extends FlutterProjectPlatform implements XcodeBasedProject {
     await _updateGeneratedXcodeConfigIfNeeded();
   }
 
+  /// Check if one the [targets] of the project is a watchOS companion app target.
+  Future<bool> containsWatchCompanion(List<String> targets) async {
+    final String bundleIdentifier = await productBundleIdentifier;
+    // A bundle identifier is required for a companion app.
+    if (bundleIdentifier == null) {
+      return false;
+    }
+    for (final String target in targets) {
+      // Create path to Info.plist file of the target.
+      final String infoFile = globals.fs.path.join(hostAppRoot.path, target, 'Info.plist');
```

Comment:
You need to check that this file actually exists.  `INFOPLIST_FILE` can be moved.

Reply:
Ok, a check has been added.